### PR TITLE
Added Hover effect on Sidebar rows

### LINKED ISSUE #1439: Add hover feedback to chat rows found in Left Hand Navigation
Problem:
The LHN on web & desktop does not have any mouse hover feedback, and thus users don’t have a clear idea of which element they are interacting with. It also makes the product feel a bit lifeless and stagnant.

Solution:
Implement a simple background color change on hover for chat rows found in the LHN. The background color for hover should use the color themeColors.componentBG which will look like this:

105834416-bafb5c80-5fca-11eb-97ac-322adcd0f3b0

There should be no hover styles added to an active chat row (the first chat row in the screenshot above)

### PR DESCRIPTION
**Pull request by parasharrajat**
Please review.

### Details
Added existing Hoverable Component as a wrapper to the OptionRow Component and based on the hover state added a style with the Background color to show the feedback.
Now we have another prop on `OptionRow` component `hoverStyle` which takes value of `styles.sidebarLinkHover` and this prop is set to `styles.modalLinkHover` on the common ancestor of all Sidebar modals(NewChat, NewGroup,SearchView) `OptionSelector`.

### Fixed Issues
Fixes #1439

### Tests
Open the app and hover over the sidebar rows to see the feedback.

### Tested On

- [x] Web
- [x] Mobile Web
- [x] Desktop
- [ ] iOS
- [ ] Android

#### Web

https://user-images.githubusercontent.com/24370807/107692744-40715300-6cd3-11eb-9c95-ca7ec2d06596.mp4

#### Desktop

https://user-images.githubusercontent.com/24370807/107692754-45ce9d80-6cd3-11eb-8a9a-39f7c51371a4.mp4

### CODE REVIEW

**Comment by shawnborton** [comment]
Looks great!

File: src/components/OptionsSelector.js [hunk 1]
```diff
@@ -184,6 +184,7 @@ class OptionsSelector extends Component {
                 </View>
                 <OptionsList
                     ref={el => this.list = el}
+                    optionHoveredStyle={[styles.modalLinkHover]}
```
Comment (roryabraham) [inline src/components/OptionsSelector.js:187]: Please refactor `styles.modalLinkHovered` and it's usages (including this one) to something more generic like `styles.hoveredComponentBG`.

Comment (roryabraham) [reply src/components/OptionsSelector.js:187]: Also, there's no need for square brackets here, you could just do:

```
optionHoveredStyle={styles.hoveredComponentBG}
```
```

File: src/pages/home/sidebar/OptionRow.js [hunk 2]
```diff
@@ -13,8 +13,12 @@ import Icon from '../../../components/Icon';
 import {Pencil, PinCircle, Checkmark} from '../../../components/Icon/Expensicons';
 import MultipleAvatars from '../../../components/MultipleAvatars';
 import themeColors from '../../../styles/themes/default';
+import Hoverable from '../../../components/Hoverable';
 
 const propTypes = {
+    // Style for hovered state
+    hoverStyle: PropTypes.arrayOf(PropTypes.object),
```
Comment (roryabraham) [inline src/pages/home/sidebar/OptionRow.js:20]:
```suggestion
    hoverStyle: PropTypes.object,
```

File: src/pages/home/sidebar/OptionRow.js [hunk 3]
```diff
@@ -38,13 +42,15 @@ const propTypes = {
 };
 
 const defaultProps = {
+    hoverStyle: [],
```
Comment (roryabraham) [inline src/pages/home/sidebar/OptionRow.js:45]:
```suggestion
    hoverStyle: styles.sidebarLinkHover,
```

File: src/pages/home/sidebar/OptionRow.js [hunk 4]
```diff
@@ -58,33 +64,37 @@ const OptionRow = ({
         : styles.sidebarLinkText;
     const textUnreadStyle = (option.isUnread || forceTextUnreadStyle)
         ? [textStyle, styles.sidebarLinkTextUnread] : [textStyle];
+    const hoveredStyle = hoverStyle ? [styles.sidebarLinkHover, ...hoverStyle] : [styles.sidebarLinkHover];
```
Comment (roryabraham) [inline src/pages/home/sidebar/OptionRow.js:67]:
```suggestion

```

File: src/pages/home/sidebar/OptionRow.js [hunk 5]
```diff
@@ -58,33 +64,37 @@ const OptionRow = ({
...
+            {hovered => (
                 <View
                     style={[
                         styles.flexRow,
                         styles.alignItemsCenter,
+                        styles.justifyContentBetween,
+                        styles.sidebarLink,
+                        styles.sidebarLinkInner,
+                        optionIsFocused ? styles.sidebarLinkActive : null,
+                        hovered && !optionIsFocused ? hoveredStyle : null,
```
Comment (roryabraham) [inline src/pages/home/sidebar/OptionRow.js:79]:
```suggestion
                        hovered && !optionIsFocused ? hoverStyle : null,
```

**Review by roryabraham — CHANGES_REQUESTED** [review]
Overall looks pretty good – just a few small changes.

**Comment by roryabraham** [comment]
Also, it looks like E2E tests are failing. They can be a bit flaky, so I would push the requested changes and if they're still failing then you might have to figure out what's going wrong

**Comment by parasharrajat** [comment]
Updated.

File: src/pages/home/sidebar/OptionRow.js [hunk 6]
```diff
@@ -93,43 +103,45 @@ const OptionRow = ({
...
-                        </View>
-                    )}
-                    {option.isPinned && (
-                        <View style={styles.ml2}>
-                            <Icon src={PinCircle} />
+                    </TouchableOpacity>
+                    {!hideAdditionalOptionStates && (
+                        <View style={styles.flexRow}>
+                            {option.hasDraftComment && (
+                            <View style={styles.ml2}>
```
Comment (roryabraham) [inline src/pages/home/sidebar/OptionRow.js:131]: Looks like this `<View>` and its children should be indented by four more spaces.

File: src/pages/home/sidebar/OptionRow.js [hunk 7]
```diff
@@ -93,43 +103,45 @@ const OptionRow = ({
...
+                                <Icon src={Pencil} />
+                            </View>
+                            )}
+                            {option.isPinned && (
+                            <View style={styles.ml2}>
```
Comment (roryabraham) [inline src/pages/home/sidebar/OptionRow.js:136]: Same with this one.

**Review by roryabraham — CHANGES_REQUESTED** [review]
I'm not sure why `optionHoveredStyle` needs to be a prop of the `OptionsList` component. It looks like that prop is only used once, and we're passing a static value of `styles.hoveredComponentBG`. So instead of:

1) Passing `styles.hoveredComponentBG` from `OptionsSelector` to `OptionsList`
2) Then passing it from `OptionsList` to `OptionRow`

We should just pass `styles.hoveredComponentBG` from `OptionsList` to `OptionRow`. Does that make sense?

**Comment by parasharrajat** [comment]
@roryabraham `OptionList` is a generic component and it is used in different locations where we need two different designs from `OptionRow`. So by default, we gave one and one we are passing from the `OptionSelector` Component which is the common ancestor.

**Review by roryabraham — APPROVED** [review]
Okay, LGTM 👍

**Pull request merged**